cmd/cody-gateway/internal/httpapi/completions/upstream.go

```diff
@@ -46,9 +46,10 @@ type upstreamHandlerMethods[ReqT UpstreamRequest] struct {
 	// validateRequest can be used to validate the HTTP request before it is sent upstream.
 	// Returning a non-nil error will stop further processing and return the given error
 	// code, or a 400.
+	// Last parameter is a boolean indicating whether the request was flagged during validation.
 	//
 	// The provided logger already contains actor context.
-	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error)
+	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, error, bool)
```

Comment:
I'd say that Go convention is to always have `error` be the last return value. `(T, bool, error)` is [quite common](https://sourcegraph.sourcegraph.com/search?q=context:sourcegraph+content:%22%2C+bool%2C+error%29%22&patternType=standard&sm=0&groupBy=repo), so I'd go with that.

```suggestion
	validateRequest func(context.Context, log.Logger, codygateway.Feature, ReqT) (int, bool, error)
```

Reply:
Fair point, looks like I'm still not super used to >2 return values.